Jenkinsfile

```diff
@@ -0,0 +1,160 @@
+pipeline {
+    agent none
+
+    stages {
+
+        stage('Build') {
+          parallel{
+            stage('Build with JDK8'){
+
+              agent {
+                docker {
+                  image 'maven:3-jdk-8-alpine'
+                  args '-v $HOME/.m2:/root/.m2'
+                }
+              }
+
+              steps {
+                sh 'mvn clean compile'
+              }
+
+            }
+
+
+            stage('Build with JDK9'){
+
+              agent {
+                docker {
+                  image 'maven:3-jdk-9-slim'
+                  args '-v $HOME/.m2:/root/.m2'
+                }
+              }
+
+              steps {
+                sh 'mvn clean compile'
+              }
+
+            }
+
+            stage('Build with JDK11'){
+
+              agent {
+                docker {
+                  image 'maven:3-jdk-11-slim'
+                  args '-v $HOME/.m2:/root/.m2'
+                }
+              }
+
+              steps {
+                sh 'mvn clean compile'
+              }
+
+            }
+
+
+
+          }
+        }
+
+	    stage('Test') {
+        parallel {
+
+          stage('Test JDK8'){
+
+            agent {
+              docker {
+                image 'maven:3-jdk-8-alpine'
+                args '-v $HOME/.m2:/root/.m2'
+
+              }
+            }
+
+            steps {
+              sh 'mvn test -PJava8'
+
+            }
+
+            post {
+              always {
+                junit 'target/surefire-reports/**/*.xml'
+               // stash includes: '**/target/coverage-reports/*', name: 'reports1'
+
+              }
+            }
+          }
+
+	        stage('Test JDK9'){
+
+            agent {
+              docker {
+                image 'maven:3-jdk-9-slim'
+                args '-v $HOME/.m2:/root/.m2'
+              }
+            }
+
+            steps {
+              sh 'mvn test -PJava9'
+
+            }
+            post {
+              always {
+                junit 'target/surefire-reports/**/*.xml'
+             //   stash includes: '**/target/coverage-reports/*', name: 'reports2'
+
+
+              }
+            }
+          }
+
+            stage('Test JDK11'){
+
+            agent {
+              docker {
+                image 'maven:3-jdk-11-slim'
+                args '-v $HOME/.m2:/root/.m2'
+              }
+            }
+
+            steps {
+              sh 'mvn test -PJava11'
+
+            }
+            post {
+              always {
+                junit 'target/surefire-reports/**/*.xml'
+               // stash includes: '**/target/coverage-reports/*', name: 'reports3'
+
+
+              }
+            }
+          }
+
+
+	       }
+		}
+
+
+
+
+
+		stage('Deploy'){
+		    when {
+			    branch 'master'
+			}
```

Comment:
We still have to add teh deployment process to maven central for master and develop as well!

Also, shouldn't this file also handle the license and style check? If yes, we should add it as well.
Btw. I want to change the style procedure from checkstyle to just using the maven plugin for auto format and the check to be consistent here and don't go through the additional configuration hustle...